cmd/kaspawallet/utils/format_kas.go

```diff
@@ -30,14 +33,36 @@ func KasToSompi(amount string) (uint64, error) {
 	parts := strings.Split(amount, ".")
 	amountStr := ""
 
+	if constants.SompiPerKaspa%10 != 0 {
+		return 0, errors.Errorf("Unable to convert to sompi when SompiPerKaspa is not a multiple of 10")
+	}
+
+	decimalPlaces := int(math.Log10(constants.SompiPerKaspa))
+	decimalStr := ""
+
 	if len(parts) == 2 {
-		amountStr = fmt.Sprintf("%s%-*s", parts[0], 8, parts[1]) // Padded with spaces at the end to fill for missing decimals: Sample "0.01234    "
-		amountStr = strings.ReplaceAll(amountStr, " ", "0")      // Make the spaces be 0s. Sample "0.012340000"
-	} else {
-		amountStr = fmt.Sprintf("%s00000000", parts[0])
+		decimalStr = parts[1]
 	}
 
+	amountStr = fmt.Sprintf("%s%-*s", parts[0], decimalPlaces, decimalStr) // Padded with spaces at the end to fill for missing decimals: Sample "0.01234    "
+	amountStr = strings.ReplaceAll(amountStr, " ", "0")                    // Make the spaces be 0s. Sample "0.012340000"
+
 	convertedAmount, err := strconv.ParseUint(amountStr, 10, 64)
 
 	return convertedAmount, err
 }
+
+func validateAmountFormat(amount string) error {
```

Comment:
change to `validateKASAmountFormat` (this function is accessible from the whole package so its name should be understandable in any context)